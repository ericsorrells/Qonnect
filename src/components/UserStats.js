// ========================================================================================
import React from 'react';
import PropTypes from 'prop-types';
// ========================================================================================
import UserStat from './UserStat';
// ========================================================================================

const UserStats = (props) => {
  const { totalEvents, openEvents, eventsAttended, followers, following } = props;
  return(
    <div className="user-stats__container">
      <UserStat stat={totalEvents}    title="Events Created"  />
      <UserStat stat={openEvents}     title="Open Events"     />
      <UserStat stat={eventsAttended} title="Events Attended"  />
      <UserStat stat={followers}      title="Followers"       />
      <UserStat stat={following}      title="Following"       />
    </div>
  )
}

UserStats.propTypes = {
  props: PropTypes.shape({
  totalEvents:    PropTypes.number,
  openEvents:     PropTypes.number,
  eventsAttended: PropTypes.number,
  followers:      PropTypes.number,
  following:      PropTypes.number
  })
};

export default UserStats;