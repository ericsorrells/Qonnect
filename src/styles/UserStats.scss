.user-stats__container {
  display: flex;
  justify-content: space-around;
  text-align: center;
  margin-left: 40px;
}
