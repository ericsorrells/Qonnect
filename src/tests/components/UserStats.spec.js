// ========================================================================================
import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import mockStoreSetup from '../../utils/mockStoreHelpers';
import 'jest-enzyme';
// ========================================================================================
import UserStats from '../../components/UserStats';
// ========================================================================================
import { basicEvent } from '../stubs/event';
import { basicProfile } from '../stubs/profile'
// ========================================================================================

describe('<UserStats/>', () => {
  it('renders correctly', () => {
    const component = shallow(
      <UserStats
        totalEvents={1}
        openEvents={2}
        eventsAttended={3}
        followers={4}
        following={5}
      />
    )

    expect(component).toMatchSnapshot();
  });
});
