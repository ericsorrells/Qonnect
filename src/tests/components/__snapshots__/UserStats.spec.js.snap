// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserStats/> renders correctly 1`] = `
<div
  className="user-stats__container"
>
  <UserStat
    stat={1}
    title="Events Created"
  />
  <UserStat
    stat={2}
    title="Open Events"
  />
  <UserStat
    stat={3}
    title="Events Attended"
  />
  <UserStat
    stat={4}
    title="Followers"
  />
  <UserStat
    stat={5}
    title="Following"
  />
</div>
`;
